src/sdg/producten/tests/test_update_catalogs.py

```diff
@@ -1,27 +1,286 @@
+from sdg.producten.models.product import Product, ProductVersie
+from sdg.producten.tests.factories.localized import LocalizedProductFactory
+from sdg.producten.tests.factories.product import (
+    ProductVersieFactory,
+    ReferentieProductFactory,
+)
+
 
 class UpdateCatalogTests(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.producten_catalogus_database = ProductenCatalogus.objects.all()
+        self.producten_database = Product.objects.all()
+        self.producten_versies_database = ProductVersie.objects.all()
+        self.localized_producten_database = LocalizedProduct.objects.all()
+
     def test_create_catalog_from_reference_if_enabled(self):
-        pass
+        ref_overheid = LokaleOverheidFactory.create(
+            automatisch_catalogus_aanmaken=False
+        )
+        ProductenCatalogusFactory.create(
+            lokale_overheid=ref_overheid,
+            is_referentie_catalogus=True,
+        )
+
+        LokaleOverheidFactory.create(
+            automatisch_catalogus_aanmaken=True,
+            organisatie__owms_end_date=None,
+        )
+
+        self.assertEqual(self.producten_catalogus_database.count(), 1)
+        call_command("update_catalogs")
+        self.assertNotEqual(self.producten_catalogus_database.count(), 1)
 
     def test_no_create_catalog_from_reference_if_disabled(self):
-        pass
+        ref_overheid = LokaleOverheidFactory.create(
+            automatisch_catalogus_aanmaken=False
+        )
+        ProductenCatalogusFactory.create(
+            lokale_overheid=ref_overheid,
+            is_referentie_catalogus=True,
+        )
+
+        LokaleOverheidFactory.create(
+            automatisch_catalogus_aanmaken=False,
+            organisatie__owms_end_date=None,
+        )
+
+        self.assertEqual(self.producten_catalogus_database.count(), 1)
+        call_command("update_catalogs")
+        self.assertEqual(self.producten_catalogus_database.count(), 1)
 
     def test_create_products_from_reference_product(self):
-        pass
+        ref_overheid = LokaleOverheidFactory.create(
+            automatisch_catalogus_aanmaken=False
+        )
+        ref_catalog = ProductenCatalogusFactory.create(
+            lokale_overheid=ref_overheid,
+            is_referentie_catalogus=True,
+        )
+        ref_product = ReferentieProductFactory.create(catalogus=ref_catalog)
+        ProductVersieFactory.create(
+            product=ref_product,
+            publicatie_datum=date.today(),
+        )
+
+        LokaleOverheidFactory.create(
+            automatisch_catalogus_aanmaken=True,
+            organisatie__owms_end_date=None,
+        )
+
+        self.assertEqual(self.producten_database.count(), 1)
+        call_command("update_catalogs")
+        self.assertEqual(self.producten_database.count(), 2)
+        self.assertEqual(
+            ref_product,
+            self.producten_database.last().referentie_product,
+        )
```

Comment:
Don't rely on ordering for comparing (`.last()`)

This test should also check if the created product has some texts copied over from the reference product, if it has a bevoegde organisatie, and that it's not published, and that it belongs to the local government you intended.

You can do:

```suggestion
        new_product = Product.objects.get(referentie_product=ref_product)
        self.assertEqual(new_product.catalog.lokale_overheid, lokale_overheid)  # needs a variable on line 77)
        self.assertEqual(new_product.current_version.specific_text, ref_product.active_version.specific_text)
```

This test should also include that some "standard texts" were copied from the reference product, to the local product.